packages/utils/src/index.ts

```diff
@@ -12,27 +12,32 @@ export const escapeRegExp = (str: string) =>
   (str || '').replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&');
 
 // { key: value, ... } => { key: mapFn(value, key), ... }
-export const mapKeys = <T, U>(
-  obj: Record<string, T>,
-  func: (value: T, key: string, obj: Record<string, T>) => U
-): Record<string, U> =>
-  Object.entries(obj).reduce((acc, [key, value]) => ({ ...acc, [key]: func(value, key, obj) }), {});
+export const mapKeys = <T, K extends keyof T, R>(
```

Comment:
This `K` type parameter isn't really useful here and the ones serving a similar since it'll also be keyof T, never more specific. It'd be better to do `T extends Record<string, any>` and when you need the key, do `keyof T`